### ISSUE TITLE
Generator never hits uppercase 'A'

### ISSUE THREAD

**Issue by ADIOP55550** [bermi/password-generator#25]
I'm pretty sure that there is some kind of bug, that prevents capital A from appearing in passwords.
I've run this command: `password-generator -c -l 50 -p "[a-zA-Z]"` around 500 times and not a single 'A' appeared. 

### Here are some example results i ran into when I was examining this:
```
$ password-generator -c -l 50 -p "[Aa]"
> aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
$ password-generator -c -l 50 -p "[Aab]"
> baaabbbababababababababaaabbabababbababbabaabbaaab
$ password-generator -c -l 50 -p "[AaBb]"
> bBbBBBbBBbbabaaBbbbaabBBaBBaababBBBaBbbBaBaBabBbbB
$ password-generator -c -l 50 -p "[BAa]"
> aaBaaaaBBaaBaBBaBBBBBBBBaaBaaaBBaBBBBBaBBaBaBaaaBa
```

Moreover, it also affects other capital letters if they're only one in the pattern:
```
$ password-generator -c -l 50 -p "[aBb]"
> aaaabbaabaabbbbbbaaaabaabbbbbaaabbbbbababaabaabbab
$ password-generator -c -l 50 -p "[a-zC]"
> urjxuxmsjxpcfnssijcwwlxikgxgeqchuqeliwtyqjrnzbqxvt
$ password-generator -c -l 50 -p "[a-zAB]"
> tvkobyiyBztramssvestxsagbmzohboebshxBnxxdnlwtkiomh
$ password-generator -c -l 50 -p "[a-zABC]"
> wCfkzoxicsxomhshdibfvtpBiuraisbdyaootuBeavboaypxdc
```

---

Also, giving it one letter as pattern generates _"Maximum call stack size exceeded"_ error.
All those generate this error:
`password-generator -c -l 50 -p "[A]"`
`password-generator -c -l 50 -p "A"`
`password-generator -c -l 50 -p "[a]"`
`password-generator -c -l 50 -p "a"`

_But that may be working as intended, because of required at least 2 characters in the pattern for some randomness._

---

### And a screenshot from console:

![Screen from console](https://user-images.githubusercontent.com/31273119/62608952-2649fe00-b901-11e9-8414-eea041a12146.png)

**Comment by ADIOP55550** [comment]
There is a file containing result of 500 calls `password-generator -c -l 50 -p "[a-zA-Z]"`. No 'A's!
[result.txt](https://github.com/bermi/password-generator/files/3476204/result.txt)

**Comment by fshields** [comment]
I've run into this issue as well.  In fact, It happens even with digits.  For example:
`generatePassword(12, false, /\d/)` will never generate a password with a zero in it (despite the README showing this as an example with a "0" in the output.)

I think I've discovered more about the issue.  It seems that it is dropping the character with the lowest ASCII value from the list of character choices.  So if you run this:
`generatePassword(12, false, /AB5/)` you will get a password with only "A" and "B" characters, but never a "5" because the character "5" has a lower ASCII value than any of the other characters.

> ASCII Values:
> "5" - decimal 53, hex 35
> "A" - decimal 65, hex 41
> "B" - decimal 66, hex 42

**Comment by fshields** [comment]
I'm pretty sure that the error lies in [line 84 of /dist/password-generator.js](https://github.com/bermi/password-generator/blob/master/dist/password-generator.js#L84):
`if (value > min && value < max) {`

The return from this function is used to select a character from a zero-based array.  However, "value" can never be zero because "min" is zero; and 0 is not less than 0 (ever).

I think that line 84 should be changed to:
`if (value >= min && value < max) {`
so that the zero-ith element of the `validChars[]` array can be chosen.

**Comment by bermi** [comment]
Thanks @ADIOP55550  for reporting this and @fshields for finding the solution to the bug. I was able to confirm it with a test case and pushed a fix to npm as 2.2.2
